core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerConfig.java

```diff
@@ -96,12 +102,12 @@ String getFieldValue() {
    */
   Set<String> getOutputFields() {
     Set<String> fields = new HashSet<String>();
```

Comment:
this actually needs to be a LinkedHashSet, or you need to return a List.  Otherwise, when you build the output schema from this set, there is no guarantee what order the fields will come in. Different reducers could output different schemas.